R/db-odbc-mssql.R

```diff
@@ -102,34 +102,46 @@
 }
 
 #' @export
-`db_compute.Microsoft SQL Server` <- function(
-                con,
-                table,
-                sql,
-                temporary = TRUE,
-                unique_indexes = list(),
-                indexes = list(),
-                ...) {
-
-
-  if (substr(table, 1, 1) != "#") {
+`db_compute.Microsoft SQL Server` <- function(con,
+                                              table,
+                                              sql,
+                                              temporary = TRUE,
+                                              unique_indexes = list(),
+                                              indexes = list(),
+                                              ...)
+{
+  # check that name has prefixed '##' if temporary
+  if(temporary && substr(table, 1, 1) != "#"){
     table <- paste0("##", table)
     message("MS SQL temp tables names require a single or double pound sign prefix. Renaming table to: ", table)
```

Comment:
By default, `compute(foo)` with no other arguments will trip this message. Perhaps it might be good to add a check if the user explicitly speciified `temporary=TRUE`, so they don't get messages about something dplyr did by itself.

Reply:
By virtue of prefixing `##` to the table name we are really not giving a choice to the end-user to set the temporary flag, I wonder if we should just remove it as an argument.  Besides, `compute()` is meant to create temporary tables.

Reply:
Prefixing ## is done only if `temporary` is TRUE. This can be changed to FALSE in the call to `compute`.

The issue of whether to have `temporary` as an arg to `compute` is probably too broad for this PR, but:
- There is a need to be able to create persistent tables, not just temporary ones
- `compute` seems as good a place to do it as any
- People are already using `compute` in this way, so removing the option would be a breaking change in the API
- You want the R-level interface to SQL Server to be as consistent as possible with other databases; implementation details like # to signify a temp table should be invisible to the user.